lib/facade_converter.ts

```diff
@@ -354,6 +396,48 @@ export class FacadeConverter extends base.TranspilerBase {
     },
   };
 
+  private es6Promises: ts.Map<CallHandler> = {
+    'Promise.resolve': (c: ts.CallExpression, context: ts.Expression) => {
+      this.emit('Future.value(');
+      this.visitList(c.arguments);  // Should only be 1.
+      this.emit(')');
+    },
+    'Promise.reject': (c: ts.CallExpression, context: ts.Expression) => {
+      this.emit('Future.error(');
+      this.visitList(c.arguments);  // Should only be 1.
+      this.emit(')');
+    },
+    'Promise': (c: ts.CallExpression, context: ts.Expression) => {
+      if (c.kind != ts.SyntaxKind.NewExpression) {
+        return true;
+      }
+      this.assert(c.arguments.length == 1);
+      const callback = <ts.ArrowFunction>(c.arguments[0]);
```

Comment:
this seems like a risky cast - might be a function, as you assert below, right? Do you need an `ArrowFunction`, or some super type?